lib/kube_link_generator.rb

```diff
@@ -86,43 +75,9 @@ def get_pods_for_role(role_name, job, options = {})
     end
   end
 
-  def patch_pod_with_imported_properties(role_name, job_name, digest)
-    client.patch_pod(
-      ENV['HOSTNAME'],
-      { metadata: { annotations: { :"skiff-in-props-#{role_name}-#{job_name}" => digest } } },
-      namespace
-    )
-  end
-
-  def get_exported_properties(role_name, pod, job_name)
-    # Exported properties are stored in a secret linked to the pod by naming convention.
-    begin
-      secret = client.get_secret("#{pod.metadata.name}-#{pod.metadata.uid}", namespace)
-    rescue
-    end
-
-    if !secret.nil?
-      digest = secret.data["skiff-exported-digest-#{job_name}"]
-      # digest not being set only happens during the spec tests???
-      if digest
-        # Copy the digest over, so that if the source role changes we can be restarted.
-        patch_pod_with_imported_properties(role_name, job_name, Base64.decode64(digest))
-      end
-      JSON.parse(Base64.decode64(secret.data["skiff-exported-properties-#{job_name}"]))
-
-    # Older implementation stored exported properties in annotations (one per job).
-    elsif pod.metadata.annotations["skiff-exported-properties-#{job_name}"]
-      # digest not being set only happens during the spec tests???
-      if pod.metadata.annotations["skiff-exported-digest-#{job_name}"]
-        # Copy the digest over, so that if the source role changes we can be restarted.
-        digest = pod.metadata.annotations["skiff-exported-digest-#{job_name}"]
-        patch_pod_with_imported_properties(role_name, job_name, digest)
-      end
-      JSON.parse(pod.metadata.annotations["skiff-exported-properties-#{job_name}"])
-
-    else
-      {}
-    end
+  def get_exported_properties(role_name, job_name)
+    secret = client.get_secret(role_name, namespace)
+    JSON.parse(Base64.decode64(secret.data["skiff-exported-properties-#{job_name}"]))
```

Comment:
This will blow up if the secret is missing right? (Assuming that's expected, document something in a comment?)